### PR TITLE
feat(CtUnresolvedImport): keep all imports in noClassPath mode

### PR DESCRIPTION
**Pull request by nharrand**
Related to #2927 .

### CODE REVIEW

**Comment by nharrand** [comment]
Ready for reviews!

File: spoon-pom/pom.xml [hunk 1]
```diff
@@ -274,12 +274,14 @@
                     <configuration>
                         <header>LICENSE-short.txt</header>
                         <includes>
-                            <include>src/main/java/**</include>
```
Comment (monperrus) [inline spoon-pom/pom.xml]: that results in deactivating license check (basedir is not src)

File: src/main/java/spoon/experimental/CtUnresolvedImport.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
+/**
```
Comment (monperrus) [inline src/main/java/spoon/experimental/CtUnresolvedImport.java]: should be the new dual license

File: src/main/java/spoon/reflect/declaration/CtImportKind.java [hunk 3]
```diff
@@ -10,5 +10,6 @@
 	ALL_TYPES,
 	ALL_STATIC_MEMBERS,
 	FIELD,
-	METHOD
+	METHOD,
+	UNRESOLVED
```
Comment (monperrus) [inline src/main/java/spoon/reflect/declaration/CtImportKind.java]: some documentation of the meaning of the new enum value?

File: src/main/java/spoon/reflect/factory/CoreFactory.java [hunk 4]
```diff
@@ -529,6 +529,11 @@ BodyHolderSourcePosition createBodyHolderSourcePosition(
 	 */
 	CtImport createImport();
 
+	/**
+	 * Creates an unresolved import.
+	 */
+	CtImport createUnresolvedImport();
```
Comment (monperrus) [inline src/main/java/spoon/reflect/factory/CoreFactory.java]: what does this mean for the end user?

File: src/test/java/spoon/test/jdtimportbuilder/ImportBuilderTest.java [hunk 5]
```diff
@@ -103,7 +105,8 @@ public void testInternalImportWhenNoClasspath() {
 
 		CtClass classA = spoon.getFactory().Class().get("it.feio.android.omninotes.models.Attachment");
 		CompilationUnit unitA = spoon.getFactory().CompilationUnit().getMap().get(classA.getPosition().getFile().getPath());
-		assertTrue(unitA.getImports().isEmpty());
+		assertTrue(unitA.getImports().stream().filter(i -> !(i instanceof CtUnresolvedImport)).collect(Collectors.toList()).isEmpty());
+		assertEquals(1, unitA.getImports().size());
```
Comment (monperrus) [inline src/test/java/spoon/test/jdtimportbuilder/ImportBuilderTest.java]: assert in the import value?

**Comment by monperrus** [comment]
cool stuff, thanks! see my comments.

File: spoon-pom/pom.xml (hunk 1 above)
Comment (nharrand) [reply spoon-pom/pom.xml]: Okay! Now I understand why I struggled so much... :-( My local copy was out of sync (I had basedir=src).

**Comment by nharrand** [comment]
Related to #2933

**Comment by nharrand** [comment]
The current implementation should work BUT it might lead to overlapping import when pretty-printing.
An example is `src/test/java/spoon/test/prettyprinter/testclasses/Rule.java`.
When imports are computed, 
```
import java.util.Arrays;
import java.util.Collections;
import java.util.Comparator;
import java.util.HashSet;
import java.util.List;
import java.util.Set;
```
are added (as these types are indeed used) but the import overlap with the pre existing CtUnresolvedImport `import java.util.*;`...

**Comment by nharrand** [comment]
OK, so now (with this PR), in noClassPath mode:
 1. Unresolved import in the original sources will remain when the model is pretty printed.
 2. But that means that import computation will never remove them even if they are unused.

This mean, that if an `import my.package.*` is unresolved we don't get the nice clean up effect that we use to have.

**Comment by monperrus** [comment]
Equals is not tested https://coveralls.io/builds/23377168/source?filename=src%2Fmain%2Fjava%2Fspoon%2Fexperimental%2FCtUnresolvedImport.java#L73

This is dangerous, could you add one assertTrue and one assertFalse?

**Comment by monperrus** [comment]
Great feature for the noClasspath mode. Thanks!

**Pull request merged**